File: MiniDOMOffice/Presenze/OreLavorate/PeriodiLavorati.vb
Imports minidom.Anagrafica
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Office
Imports minidom.Internals


Namespace Internals


    Public NotInheritable Class CPeriodiLavorati
        Inherits CModulesClass(Of PeriodoLavorato)

        Friend Sub New()
            MyBase.New("modOfficePeriodiLavorati", GetType(PeriodoLavoratoCursor), 0)
        End Sub


    End Class
End Namespace

Partial Class Office

    Private Shared m_PeriodiLavorati As CPeriodiLavorati = Nothing

    Public Shared ReadOnly Property PeriodiLavorati As CPeriodiLavorati
        Get
            If (m_PeriodiLavorati Is Nothing) Then m_PeriodiLavorati = New CPeriodiLavorati
            Return m_PeriodiLavorati
        End Get
    End Property

End Class

File: MiniDOMOffice/Presenze/OreLavorate/PeriodiLavorati.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;

namespace minidom
{
    namespace repositories
    {

        /// <summary>
        /// Repository di <see cref="PeriodoLavorato"/>
        /// </summary>
        [Serializable]
        public sealed class CPeriodiLavorati 
            : CModulesClass<PeriodoLavorato>
        {

            /// <summary>
            /// Costruttore
            /// </summary>
            public CPeriodiLavorati() 
                : base("modOfficePeriodiLavorati", typeof(PeriodoLavoratoCursor), 0)
            {
            }
        }
    }

    public partial class Office
    {
        private static CPeriodiLavorati m_PeriodiLavorati = null;

        /// <summary>
        /// Repository di <see cref="PeriodoLavorato"/>
        /// </summary>
        public static CPeriodiLavorati PeriodiLavorati
        {
            get
            {
                if (m_PeriodiLavorati is null)
                    m_PeriodiLavorati = new CPeriodiLavorati();
                return m_PeriodiLavorati;
            }
        }
    }
}